### PR TITLE
feat(Other): Add Zephyr hwinfo driver's dependencies

### PR DESCRIPTION
**Pull request by hfakkiz**
### Description

This PR adds Zephyr hwinfo driver's dependencies to zephyr_library_sources.

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Review by sihyung-maxim — APPROVED** [review]
Will merge after Sadik and Maureen approves.

**Review by MaureenHelm — APPROVED** [review]

**Review by ozersa — APPROVED** [review]

**Pull request merged**
